client/lib/analytics/index.js

```diff
@@ -287,20 +283,28 @@ const analytics = {
 				return;
 			}
 
+			let actualProperties = {};
+
 			if ( _superProps ) {
 				_dispatch && _dispatch( { type: ANALYTICS_SUPER_PROPS_UPDATE } );
-				superProperties = _superProps.getAll( _selectedSite, _siteCount );
-				eventProperties = assign( {}, eventProperties, superProperties ); // assign to a new object so we don't modify the argument
+				const superProperties = _superProps.getAll( _selectedSite, _siteCount );
```

Comment:
Rather than allowing all of the site-specific super props to be added, and then removing just the blog ID, could we instead pass `eventProperties` as a third param to `superProps.getAll()`, do the check there, and not add the site-specific super props at all if it fails? That might keep the stats collected more consistent.

Reply:
I agree with @kwight. Another option would be using `shouldReportOmitBlogId` to instead toggle whether we pass through `_selectedSite`

For example:
```jsx
const site = shouldReportOmitBlogId( eventProperties.path ) ? _selectedSite : null;
const superProperties = _superProps.getAll( site, _siteCount );
```

Reply:
@Copons pick whichever approach you please, and give give us a ping for us to do another round of testing. I think the overall approach here is solid 👍

Reply:
Ooh, great suggestions thanks!
I'm gonna go with @gwwar's approach since I'd rather leave `superProps.getAll` alone in the off chance we actually want all available super props no matter what.
Also, well, its `selectedSite` parameter seems to be there almost exactly for this particular purpose! 🙂